### PR TITLE
Feature/kitosudv 2114 task 2159

### PR DESCRIPTION
**Pull request by mrjsawdk**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [x] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [x] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Core.ApplicationServices/GDPR/Write/DataProcessingRegistrationWriteService.cs [hunk 1]
```diff
@@ -112,9 +212,56 @@ public Maybe<OperationError> Delete(Guid dataProcessingRegistrationUuid)
...
+           Func<DataProcessingRegistration, int, Result<TAssignment, OperationError>> assign,
+           Func<DataProcessingRegistration, int, Result<TAssignment, OperationError>> unAssign)
+           where TAssignment : class, IHasId, IHasUuid
+        {
+            var newUuids = assignedItemUuid.Match(uuids => uuids.ToList(), () => new List<Guid>());
+            if (newUuids.Distinct().Count() != newUuids.Count)
+            {
+                return new OperationError($"Duplicates of '{subject}' are not allowed", OperationFailure.BadInput);
+            }
+            var existingAssignemnts = getExistingState(dpr).ToDictionary(x => x.Uuid);
```
Comment (JacobMalling) [inline Core.ApplicationServices/GDPR/Write/DataProcessingRegistrationWriteService.cs:234]: existingAssignemnts => existingAssingments

File: Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/YesNoIrrelevantMappingExtensions.cs [hunk 2]
```diff
@@ -16,25 +16,25 @@ static YesNoIrrelevantMappingExtensions()
...
             ApiToDataMap = new Dictionary<YesNoIrrelevantChoice, YesNoIrrelevantOption>
             {
-                { YesNoIrrelevantChoice.Yes, YesNoIrrelevantOption.YES },
-                { YesNoIrrelevantChoice.No, YesNoIrrelevantOption.NO },
-                { YesNoIrrelevantChoice.Irrelevant, YesNoIrrelevantOption.IRRELEVANT },
-                { YesNoIrrelevantChoice.Undecided, YesNoIrrelevantOption.UNDECIDED }
+                { YesNoIrrelevantChoice.Yes ,YesNoIrrelevantOption.YES},
+                { YesNoIrrelevantChoice.No ,YesNoIrrelevantOption.NO},
+                { YesNoIrrelevantChoice.Irrelevant,YesNoIrrelevantOption.IRRELEVANT},
+                { YesNoIrrelevantChoice.Undecided,YesNoIrrelevantOption.UNDECIDED}
```
Comment (JacobMalling) [inline Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/YesNoIrrelevantMappingExtensions.cs:22]: Der er sket noget med formatteringen her.

File: Tests.Unit.Core.ApplicationServices/ApplicationServices/GDPR/DataProcessingRegistrationWriteServiceTest.cs [hunk 3]
```diff
@@ -248,6 +238,802 @@ public void Cannot_Update_Name_If_Dpr_Resolution_Fails()
...
+            };
+            var transaction = ExpectTransaction();
+            var orgDbId = A<int>();
+
+            ExpectIfUuidHasValueResolveIdentityDbIdReturnsId<Organization>(organizationUuid, orgDbId);
+            ExpectCreateDataProcessingRegistrationReturns(orgDbId, parameters, parameters.Name.NewValue, createdRegistration);
+            return (organizationUuid, parameters, createdRegistration, transaction);
+        }
+
+        //TODO: Update edge cases
```
Comment (JacobMalling) [inline Tests.Unit.Core.ApplicationServices/ApplicationServices/GDPR/DataProcessingRegistrationWriteServiceTest.cs:1035]: Hvad med denne TODO?

**Review by JacobMalling — APPROVED** [review]
Mindre rettelser. Fint med AllExcept tilføjelsen til enum tests

File: Tests.Unit.Core.ApplicationServices/ApplicationServices/GDPR/DataProcessingRegistrationWriteServiceTest.cs (hunk 3 above)
Comment (mrjsawdk) [reply Tests.Unit.Core.ApplicationServices/ApplicationServices/GDPR/DataProcessingRegistrationWriteServiceTest.cs:1035]: er håndteret men den gemte sig
